### DISCUSSION TITLE
Search in function or type

### DISCUSSION THREAD

**Discussion by Ha5hBr0wn** [helix-editor/helix#7187] [Q&A] [answered]
We can search a file with ''/" and the global workspace with "<space>/"

Is there a way to search within the current function or class? Seems like something that is possible with the tree sitter grammars?

**Comment by gabydd** [comment, marked as answer]
You can't really search within a function but you can select a function with ``maf`` (``mat`` for class) and then use ``s`` to select the regex matches of your search. I guess the only plus of "function" specific search over going to the start of the function then searching, would letter still shows results if the search term is only outside of he function. In my opinion that isn't really worth it though.
